cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java

```diff
@@ -636,12 +636,15 @@ private CloudVmTypes getCloudVmTypes(CloudCredential cloudCredential, Region reg
         Map<String, Set<VmType>> cloudVmResponses = new HashMap<>();
         Map<String, VmType> defaultCloudVmResponses = new HashMap<>();
 
-        for (AvailabilityZone availabilityZone : regions.getCloudRegions().get(region)) {
-            Set<VmType> types = vmTypes.get(region).stream()
-                    .filter(enabledInstanceTypeFilter)
-                    .collect(Collectors.toSet());
-            cloudVmResponses.put(availabilityZone.value(), types);
-            defaultCloudVmResponses.put(availabilityZone.value(), defaultVmTypes.get(region));
+        List<AvailabilityZone> availabilityZones = regions.getCloudRegions().get(region);
+        if (availabilityZones != null && !availabilityZones.isEmpty()) {
+            for (AvailabilityZone availabilityZone : availabilityZones) {
+                Set<VmType> types = vmTypes.get(region).stream()
+                        .filter(enabledInstanceTypeFilter)
+                        .collect(Collectors.toSet());
+                cloudVmResponses.put(availabilityZone.value(), types);
+                defaultCloudVmResponses.put(availabilityZone.value(), defaultVmTypes.get(region));
+            }
         }
```

Comment:
Should we throw an exception, if the AZ list is empty? If not, at least log the fact please.

Reply:
Are the AZs required? Are you sure that we do not break the process with exception?

Reply:
Well, without an exception, the dropdown on the UI will be empty, which might also be a good enough clue for the user, that something is amiss.

Reply:
the log message is ok for me, I will update with it